# Louisville Free Public Library

Analysis of Young Adult (YA) genre in the Louisville Free Public Library collection.

## Questions

In this analysis we will look at the following questions:

- How much was spent on the collection for YA? 
- How many books are in the collection for YA?
- How does YA spending compare to other collections?
- Did the spending on YA change over time?
- Is YA more or less popular at any of the locations?

### Load the clean library collection data and show a preview of the data

In [2]:
import pandas as pd
import numpy as np
from pathlib import Path

# load the clean books data into a dataframe and show the first few rows
books_data_path = Path('results/books_clean.csv.gz')
books_df = pd.read_csv(books_data_path)
books_df.head()

BibNum                                              Title  \
0  707409  Jeff Immelt and the new GE way : innovation, t...   
1  707411  Robin rescues dinner : 52 weeks of quick-fix m...   
2  707411  Robin rescues dinner : 52 weeks of quick-fix m...   
3  707411  Robin rescues dinner : 52 weeks of quick-fix m...   
4  707411  Robin rescues dinner : 52 weeks of quick-fix m...   

                 Author  PublicationYear ItemType     ItemCollection  \
0   Magee, David, 1965-             2009     Book  Adult Non-Fiction   
1  Miller, Robin, 1964-             2009     Book  Adult Non-Fiction   
2  Miller, Robin, 1964-             2009     Book  Adult Non-Fiction   
3  Miller, Robin, 1964-             2009     Book  Adult Non-Fiction   
4  Miller, Robin, 1964-             2009     Book  Adult Non-Fiction   

             ItemLocation  ItemPrice        Genre Audience  
0                    Main      25.95  Non-Fiction    Adult  
1               Southwest      19.99  Non-Fiction    Adult  
2               Southwest      19.99  Non-Fiction    Adult  
3  Remote Shelving - Main      19.99  Non-Fiction    Adult  
4  Remote Shelving - Main      19.99  Non-Fiction    Adult

### How much was spent on the collection for YA?

In [6]:
# TODO: First figure out which records in the dataframe are YA using a mask
# YA = Genre: Fiction, Audience = Teen. Then slice the DataFrame using the mask 
# and sum the ItemPrice column and format the result.

ya_mask = (books_df['Audience']=='Teen') & (books_df['Genre'] == 'Fiction')
ya_mask

"${:,}".format(books_df[ya_mask]['ItemPrice'].sum())

'$555,691.26'

The YA collection has a total cost of $555,691.26

### How many books in  the collection are YA?

In [9]:
# TODO: Create a new column in the dataframe called YA_Category, use 
# value_counts() to get the count and percent of YA books, and use the concat()
# function to combine the counts and percents into a single dataframe. 

books_df['YA_Category'] = np.where(ya_mask, 'YA', 'Other')

ya_counts = books_df['YA_Category'].value_counts().apply(lambda x: "{:,}".format(x))
ya_percents = books_df['YA_Category'].value_counts(normalize=True).mul(100).round(1).astype(str) + '%'

pd.concat([ya_counts, ya_percents], axis=1, keys=['books','percentage'])

books percentage
Other  1,145,946      96.5%
YA        41,257       3.5%

YA accounts for 3.5% percent of the total number of books in the collection.

### How does YA spending compare to other collections?

In [10]:
# TODO: Group the data by Genre and Audience using groupby() and use sum() to 
# get the total cost. Format the totals as currency.

books_df.groupby(['Genre','Audience'])['ItemPrice'].sum().apply(lambda x: "${:,.2f}".format(x))

Genre        Audience
Fiction      Adult       $3,457,835.27
             Children      $687,553.59
             Teen          $555,691.26
             Unknown     $1,731,767.36
Non-Fiction  Adult       $9,209,529.31
             Children    $1,597,204.37
             Teen          $401,104.39
             Unknown       $875,794.32
Unknown      Adult         $281,617.43
             Children    $2,505,961.73
             Teen              $119.09
             Unknown       $533,619.31
Name: ItemPrice, dtype: object

### Did the spending on YA change over time?

In [19]:
# TODO: Calculate the counts (value_counts()), total cost (sum()), and average 
# cost (mean()) for all YA books by publication year using groupby(). 
# Concatenate the counts and costs using concat() into a single dataframe. 
# Format the counts and costs as numbers and currency.

ya_years_count = books_df[['PublicationYear', 'ItemPrice']][books_df['YA_Category']=='YA'].groupby('PublicationYear').count()
ya_years_count.columns = ['BookCount']

ya_years_total = books_df[['PublicationYear', 'ItemPrice']][books_df['YA_Category']=='YA'].groupby('PublicationYear').sum()
ya_years_total.columns = ['TotalCost']

ya_years_average = books_df[['PublicationYear', 'ItemPrice']][books_df['YA_Category']=='YA'].groupby('PublicationYear').mean()
ya_years_average.columns = ['AverageCost']

ya_years_summary = pd.concat([ya_years_count, ya_years_total, ya_years_average], axis=1)

ya_years_summary['BookCount'] = ya_years_summary['BookCount'].apply(lambda x: "{:,}".format(x))

ya_years_summary

BookCount  TotalCost  AverageCost
PublicationYear                                  
1919                    1      20.00    20.000000
1938                    3      42.53    14.176667
1939                    5     124.95    24.990000
1966                    1       2.99     2.990000
1967                    1       6.99     6.990000
1968                    2      22.94    11.470000
1970                    3      30.89    10.296667
1971                    8     126.90    15.862500
1972                    2      27.95    13.975000
1973                   15     101.13     6.742000
1974                    6      70.91    11.818333
1976                    7      66.01     9.430000
1977                    9      89.01     9.890000
1978                    8      59.71     7.463750
1979                    2      20.98    10.490000
1980                    4      60.58    15.145000
1981                    4      40.92    10.230000
1982                    1      13.49    13.490000
1983                    3      11.57     3.856667
1984                    9      57.87     6.430000
1985                   14     141.15    10.082143
1986                    1       6.99     6.990000
1987                   15     145.03     9.668667
1988                    6      48.96     8.160000
1989                    8      85.56    10.695000
1990                    8      74.43     9.303750
1991                   13     131.82    10.140000
1992                   20     150.04     7.502000
1993                   41     336.98     8.219024
1994                   49     575.11    11.736939
1995                   15     155.69    10.379333
1996                   57     576.40    10.112281
1997                   85    1122.03    13.200353
1998                   54     577.92    10.702222
1999                  117    1392.21    11.899231
2000                  113    1404.27    12.427168
2001                  156    1824.02    11.692436
2002                  153    1868.17    12.210261
2003                  214    2636.94    12.322150
2004                  374    4228.69    11.306658
2005                  523    6984.28    13.354264
2006                  919   11682.18    12.711839
2007                1,185   14572.44    12.297418
2008                1,304   17444.92    13.378006
2009                1,209   16679.89    13.796435
2010                1,577   21765.42    13.801788
2011                1,889   28039.02    14.843314
2012                2,118   31727.66    14.980009
2013                2,732   40770.55    14.923335
2014                3,262   50246.38    15.403550
2015                3,371   52836.41    15.673809
2016                2,916   47197.27    16.185621
2017                2,981   46134.20    15.476082
2018                3,483   36891.70    10.591932
2019                2,748   28427.17    10.344676
2020                2,790   28231.59    10.118849
2021                2,621   26839.37    10.240126
2022                1,987   30215.27    15.206477
2023                   35     524.81    14.994571

Spending on YA books - there is an increase in how much has been spent on YA books over the past 15 years, with a steady increase since 2002.

Is there a correlation between ItemPrice and PublicationYear?

In [20]:
# TODO: Use the corr() function to determine if there is a correlation between
# the ItemPrice and PublicationYear.

books_df[['ItemPrice','PublicationYear']].corr()

ItemPrice  PublicationYear
ItemPrice         1.000000        -0.256709
PublicationYear  -0.256709         1.000000

Price goes down after time. 

### Is YA more or less popular at any of the locations?

In [27]:
# TODO: Calculate the total number of books by location and the number of YA
# books by location and concatenate them into a single DataFrame. Add a new column
# to show the % of books by location that are YA. Format the values appropriately.

location_ya = books_df['ItemLocation'][books_df['YA_Category'] == 'YA'].value_counts()
location_ya.rename("YABookCount", inplace=True)

location_all = books_df['ItemLocation'].value_counts()
location_all.rename("TotalBookCount", inplace=True)

location_summary = pd.concat([location_all, location_ya], axis=1)

location_perc = location_ya/location_all
location_perc.rename("PercTotal", inplace=True)
location_perc = location_perc.map(lambda x: '{:.2%}'.format(x))

location_summary = pd.concat([location_all, location_ya, location_perc], axis=1)

location_summary

TotalBookCount  YABookCount PercTotal
Remote Shelving - Main             139969         26.0     0.02%
Northeast                          124339       4928.0     3.96%
Southwest                          121914       5413.0     4.44%
Main                               120742         12.0     0.01%
South Central                      115614       5238.0     4.53%
Bon Air                             74551       2072.0     2.78%
St Matthews                         69417       2636.0     3.80%
Jeffersontown                       56620       1425.0     2.52%
Iroquois                            52190       1910.0     3.66%
Highlands - Shelby Park             45352       1457.0     3.21%
Crescent Hill                       42683       1252.0     2.93%
Childrens Main Library              38668          1.0     0.00%
Middletown                          33069       1333.0     4.03%
Shively                             23549       1206.0     5.12%
Newburg                             23536       1011.0     4.30%
Fairdale                            23025        983.0     4.27%
Shawnee                             22861       1830.0     8.00%
Western                             21615       1009.0     4.67%
Portland                            13313        491.0     3.69%
Childrens Bookmobile                 9092        187.0     2.06%
Remote Shelving - Shawnee            9060       2988.0    32.98%
Main Teen                            6018       3848.0    63.94%
Content Management                      4          1.0    25.00%
Adult Bookmobile                        2          NaN      nan%

Main Teen has the most teen books by percent, which makes a lot of sense. Following that its Shawnee by a bit. The one that has the least amount of YA books by percent is J-town (this is exluding the main branch, which has their books already broken out by section, and the remote shelving units and childrens bookmobile).